### PR TITLE
rustup https://github.com/rust-lang/rust/pull/71116

### PR DESCRIPTION
**Pull request by matthiaskrgr**
changelog: none

### CODE REVIEW

File: clippy_lints/src/utils/author.rs [hunk 1]
```diff
@@ -93,11 +93,21 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Author {
...
     fn check_variant(&mut self, cx: &LateContext<'a, 'tcx>, var: &'tcx hir::Variant<'_>) {
+        use rustc_hir::def_id::{LocalDefId, CRATE_DEF_INDEX};
+        use rustc_hir::{HirId, DUMMY_ITEM_LOCAL_ID};
+
+        const DUMMY_HIR_ID: HirId = HirId {
+            owner: LocalDefId {
+                local_def_index: CRATE_DEF_INDEX,
+            },
+            local_id: DUMMY_ITEM_LOCAL_ID,
+        };
```
Comment (eddyb) [inline clippy_lints/src/utils/author.rs:104]: There must be a way to avoid reintroducing this.

File: clippy_lints/src/utils/author.rs [hunk 2]
```diff
@@ -93,11 +93,21 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Author {
...
+
         if !has_attr(cx.sess(), &var.attrs) {
             return;
         }
         prelude();
-        PrintVisitor::new("var").visit_variant(var, &hir::Generics::empty(), hir::DUMMY_HIR_ID);
+        PrintVisitor::new("var").visit_variant(var, &hir::Generics::empty(), DUMMY_HIR_ID);
```
Comment (eddyb) [inline clippy_lints/src/utils/author.rs:110]: Something like `cx.tcx.hir().get_parent(var.hir_id)` should give you a valid `HirId`.

Comment (eddyb) [reply clippy_lints/src/utils/author.rs:110]: Which probably means we can remove the `HirId` arguments from `visit_variant` and `visit_variant_data` and `visit_enum_def`, maybe. Depends on what visitors use them and for what.

Comment (matthiaskrgr) [reply clippy_lints/src/utils/author.rs]: Thanks!
I'm not familiar with how the author lint works, I switched to `cx.tcx.hir().get_parent_node(var.id);` now.

**Comment by flip1995** [comment]
LGTM 

@bors r+

File: clippy_lints/src/utils/author.rs [hunk 3]
```diff
@@ -97,7 +97,8 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Author {
             return;
         }
         prelude();
-        PrintVisitor::new("var").visit_variant(var, &hir::Generics::empty(), hir::DUMMY_HIR_ID);
+        let dummy_hir_id = cx.tcx.hir().get_parent_node(var.id);
```
Comment (eddyb) [inline clippy_lints/src/utils/author.rs:100]: This is not dummy. This is the correct `HirId` for the HIR visitor method.

Comment (matthiaskrgr) [reply clippy_lints/src/utils/author.rs:100]: Eh.. :(
I'll change it in a follow up pr that we can add to the rollup

File: clippy_lints/src/utils/author.rs [hunk 4]
```diff
@@ -97,7 +97,8 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Author {
...
+        PrintVisitor::new("var").visit_variant(var, &hir::Generics::empty(), dummy_hir_id);
```
Comment (flip1995) [inline clippy_lints/src/utils/author.rs:101]:
```suggestion
        let parent_hir_id = cx.tcx.hir().get_parent_node(var.id);
        PrintVisitor::new("var").visit_variant(var, &hir::Generics::empty(), parent_hir_id);
```

File: clippy_lints/src/utils/author.rs (hunk 3 above)
Comment (flip1995) [reply clippy_lints/src/utils/author.rs:100]: GHA got stuck anyway in the caching process.

**Pull request merged**